performance_manager/lib/logging_utils.py

```diff
@@ -18,70 +17,75 @@ def __init__(self, process_name: str, **metadata: MdValues) -> None:
+            if key in self.protected_keys:
+                continue
+            self.metadata[str(key)] = str(value)
 
     def log_start(self) -> None:
         """log the start of a proccess"""
-        self.uuid = uuid.uuid4()
+        self.default_data["uuid"] = uuid.uuid4()
+        self.default_data["process_id"] = os.getpid()
+        self.default_data["status"] = "started"
+        self.default_data.pop("duration", None)
+        self.default_data.pop("error_type", None)
+
         self.start_time = time.monotonic()
-        self.process_id = os.getpid()
-        logging_string = (
-            f"parent={self.parent}, "
-            f"start={self.process_name}, "
-            f"uuid={self.uuid}, "
-            f"process_id={self.process_id}"
-        )
-
-        if self.metadata:
-            metadata_string = ", ".join(
-                [f"{key}={value}" for (key, value) in self.metadata.items()]
-            )
-            logging_string += f", {metadata_string}"
-        logging.info(logging_string)
+
+        logging.info(self._get_log_string())
 
     def log_complete(self) -> None:
         """log the completion of a proccess with duration"""
         duration = time.monotonic() - self.start_time
-        logging_string = (
-            f"parent={self.parent}, "
-            f"complete={self.process_name}, "
-            f"uuid={self.uuid}, "
-            f"process_id={self.process_id}, "
-            f"duration={duration:.2f}"
-        )
-
-        if self.metadata:
-            metadata_string = ", ".join(
-                [f"{key}={value}" for (key, value) in self.metadata.items()]
-            )
-            logging_string += f", {metadata_string}"
-        logging.info(logging_string)
+        self.default_data["status"] = "complete"
+        self.default_data["duration"] = f"{duration:.2f}"
+
+        logging.info(self._get_log_string())
 
     def log_failure(self, exception: Exception) -> None:
-        """log the failure of a process with exception details"""
+        """log the failure of a process with exception type"""
         duration = time.monotonic() - self.start_time
-        logging_string = (
-            f"parent={self.parent}, "
-            f"failed={self.process_name}, "
-            f"uuid={self.uuid}, "
-            f"process_id={self.process_id}, "
-            f"duration={duration:.2f}, "
-            f"error_type={type(exception).__name__}"
-        )
-
-        if self.metadata:
-            metadata_string = ", ".join(
-                [f"{key}={value}" for (key, value) in self.metadata.items()]
-            )
-            logging_string += f", {metadata_string}"
-        logging.exception(logging_string)
+        self.default_data["status"] = "failed"
+        self.default_data["duration"] = f"{duration:.2f}"
+        self.default_data["error_type"] = type(exception).__name__
+
+        logging.exception(self._get_log_string())
+        logging.exception(exception)
```

Comment:
always log `exception` traceback for de-bugging. were we handing this a different way before?

Reply:
i think if you use `logging.exception(message)` it will log the traceback. thats the difference between `logging.error` and `logging.exception`. as written, i think we'll log the traceback twice.

the bigger problem is splunk wants to separate all of the tracebacks by `\n`, which i think is a problem other teams are also dealing with.